### PR TITLE
INTMDB-694: Add PrivateEndpoint.srvShardOptimizedConnectionString to cluster

### PR DESCRIPTION
**Pull request by maastha**
## Description

Add support for PrivateEndpoint.srvShardOptimizedConnectionString to cluster

_Jira ticket_: [INTMDB-694](https://jira.mongodb.org/browse/INTMDB-694)

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Testing
Created a sharded cluster with Terraform and outputted the the new field `srvShardOptimizedConnectionString` by:
```
  private_endpoints = flatten([for cs in mongodbatlas_cluster.aws_private_connection.connection_strings : cs.private_endpoint])


connection_strings_shard_optimized = [
    for pe in local.private_endpoints : pe.srv_shard_optimized_connection_string
    if contains([for e in pe.endpoints : e.endpoint_id], aws_vpc_endpoint.vpce_east.id)
  ]
```
Screenshot 2023-05-04 at 11 52 24

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 1]
```diff
@@ -1535,10 +1535,11 @@ func flattenPrivateEndpoint(privateEndpoints []matlas.PrivateEndpoint) []map[str
...
 		endpoints = append(endpoints, map[string]interface{}{
-			"connection_string":     endpoint.ConnectionString,
-			"srv_connection_string": endpoint.SRVConnectionString,
-			"endpoints":             flattenEndpoints(endpoint.Endpoints),
-			"type":                  endpoint.Type,
+			"connection_string":                     endpoint.ConnectionString,
+			"srv_connection_string":                 endpoint.SRVConnectionString,
+			"srv_shard_optimized_connection_string": endpoint.SRVShardOptimizedConnectionString,
+			"endpoints":                             flattenEndpoints(endpoint.Endpoints),
+			"type":                                  endpoint.Type,
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_cluster.go:1542]: Do you need to change `mongodbatlas_advanced_cluster` too?

**Review by martinstibbe — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_cluster.go (hunk 1 above)
Comment (maastha) [reply mongodbatlas/resource_mongodbatlas_cluster.go:1542]: _advanced_cluster resources use the same methods to read/flatten connectionStrings.PrivateEndpoint so these changes will propagate to `mongodbatlas_advanced_cluster` as well

Comment (andreaangiolillo) [reply mongodbatlas/resource_mongodbatlas_cluster.go:1542]: Thanks for the clarification!

**Pull request merged**